### PR TITLE
Recover 'damaged tent' if tent is damaged (fix #19528)

### LINKED ISSUE #19528: Damaged tents can be taken down, set up as fully repaired tents
One can come across damaged tents missing walls in field campsites. If it still has a groundsheet tile intact, one can take down the tent to have a fully repaired tent available.
(Note that you can see the unrelated issue #15569 present in these screenshots as well)

[save.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/616547/save.zip)

screen shot 2016-11-28 at 9 25 10 am
screen shot 2016-11-28 at 9 25 26 am

### PR DESCRIPTION
**Pull request by BrettDong**
Fixes #19528.

### CODE REVIEW

File: src/iexamine.cpp [hunk 1]
```diff
@@ -810,6 +810,40 @@ void iexamine::portable_structure(player &p, const tripoint &examp)
...
+        furn_id door =
+            dropped == "tent_kit" ? f_canvas_door : f_large_canvas_door;
+        furn_id door_opened =
+            dropped == "tent_kit" ? f_canvas_door_o : f_large_canvas_door_o;
+        furn_id center_floor =
+            dropped == "large_tent_kit" ? f_center_groundsheet : floor;
+        for( int i = -radius; i <= radius; i++ ) {
+            for( int j = -radius; j <= radius; j++ ) {
+                if( i != -radius && i != radius && j != -radius && j != radius ) {
+                    if( g->m.furn( examp.x + i, examp.y + j ) != floor && g->m.furn( examp.x + i, examp.y + j ) != center_floor ) {
```
Comment (keyspace) [inline src/iexamine.cpp]: Seems to be accessing the same tile, could use a local var (`const furn_id&`?..). Same for the "walls" check below.

A very terse block, could use a few comments for readability. "Loop through all tiles in radius" for the `for`s above, "Check floor tiles" and "Check wall tiles" for the `if-else`... Miight not be necessary, though, if local vars are used.

**Comment by Coolthulhu** [comment]
Would be much cleaner as a map:
```
struct tent_data {
string floor;
string wall;
string door;
...
};
```

Then you wouldn't have those ugly `dropped == "tent_kit"` checks.

**Comment by BrettDong** [comment]
Those codes are copied from `iuse::portable_structure`. Store those tents in a `struct` is better, but I'll do this in a later PR, not in this.

**Comment by Coolthulhu** [comment]
Few general tent notes:
* There is a lot of tent data provided in the json of tents. Those could be reused here
* Name of the tent is not translated

**Comment by BrettDong** [comment]
> Name of the tent is not translated

This is solved in #20156 .

**Pull request merged**